### ISSUE TITLE
KeyError: 'state_dict'     not fix

### ISSUE THREAD

**Issue by 1301715759** [d8ahazard/sd_dreambooth_extension#9]
![图片](https://user-images.githubusercontent.com/44659174/200397504-90027967-3ea5-4b21-947a-e718bb868ae8.png)
![图片](https://user-images.githubusercontent.com/44659174/200397986-35d16bc3-3673-474a-8820-a7af0edf734c.png)

created with animefull-latest.ckpt model [7G ver].
fact that  always find this problem in dreambooth.
I tried to solved it like this, in some scripts it works,might can not work on this scripts
![图片](https://user-images.githubusercontent.com/44659174/200398603-989071e7-5b42-45e2-af76-9c188cd7a829.png)

**Comment by d8ahazard** [comment]
Just pushed f033e79 that should hopefully address this.

**Comment by 1301715759** [comment]
> Just pushed [f033e79](https://github.com/d8ahazard/sd_dreambooth_extension/commit/f033e7900b16540dda60d145b7e129d6a712b519) that should hopefully address this.
still have porblems create model with animefull-latest.ckpt [e6e8e1fc]
![图片](https://user-images.githubusercontent.com/44659174/200495429-1423e9b1-00c6-4766-b081-3329d3986731.png)

**Comment by d8ahazard** [comment]
So, this one is tricky. 

I can't find that checkpoint directly, so I can't look at why it's breaking, but it seems as if the previous fix in trying to load the checkpoint from a different path doesn't help. If you can provide a URL to the ckpt, I can look at the contents and try to determine why it's failing.

**Comment by 1301715759** [comment]
@d8ahazard 
https://pub-2fdef7a2969f43289c42ac5ae3412fd4.r2.dev/animefull-latest.ckpt 
fact this model was nice with training on dreambooth for colab. I hop this problem could be solved.

**Comment by CrazyBoyM** [comment]
I think it is caused by that you have use a special ckpt model file which doesn't has a 'state_dict' key", and the follow code may help you:
```
  sd = torch.load(model_path, map_location="cpu")
  if "state_dict" not in sd:
      pruned_sd = {
          "state_dict": dict(),
      }
  else:
      pruned_sd = dict()
  print(sd.keys())
  for k in sd.keys():
      if k != "optimizer_states":
          if "state_dict" not in sd:
              pruned_sd["state_dict"][k] = sd[k]
          else:
              pruned_sd[k] =sd[k]
```
just remove "optimizer_states" key, and put other key to "state_dict" to get a normal model for you.

**Comment by 1301715759** [comment]
> 

Oh thanks,it works well.

**Comment by 1301715759** [comment]
> I think it is caused by that you have use a special ckpt model file which doesn't has a 'state_dict' key", and the follow code may help you:
> 
> ```
>   sd = torch.load(model_path, map_location="cpu")
>   if "state_dict" not in sd:
>       pruned_sd = {
>           "state_dict": dict(),
>       }
>   else:
>       pruned_sd = dict()
>   for k in sd.keys():
>       if k != "optimizer_states":
>           if "state_dict" not in sd:
>               pruned_sd["state_dict"][k] = sd[k]
>           else:
>               pruned_sd[k] = sd[k]
>   torch.save(pruned_sd, "model-pruned.ckpt")
> ```
> 
> just remove "optimizer_states" key, and put other key to "state_dict" to get a normal model for you. So that the code:
> 
> ```
> checkpoint = torch.load(checkpoint_file[0])['state_dict']
> ```
> 
> may work well.

Oh thanks,it works well

**Comment by d8ahazard** [comment]
I'll add this to the code soon, unless someone wants to submit a PR.

**Comment by tetsuoo-online** [comment]
I'm having the same error messages as 1301715759 user, except that I'm using sd-v1-4.ckpt, which is the default model.
I have no idea how to implement the solution proposed by CrazyBoyM, and I know every time I try to chanfge code from a file I'll get in troubles later (things like : "that file is no good you need to commit first blabla" and then I can't update properly again)

**Comment by d8ahazard** [comment]
Closing, fixed with a9bbe70
